### PR TITLE
Pec globus

### PR DESCRIPTION
**Pull request by danlu1**
This PR includes:
1. A config yaml to provision the S3 and grant synapse and Dan's sandbox account access for now. 

depends on https://github.com/Sage-Bionetworks/aws-infra/pull/349

### CODE REVIEW

File: config/prod/PsychENCODE-Globus-S3.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Provision a Synapse External Bucket (http://docs.synapse.org/articles/custom_storage_location.html)
+template:
+  type: "http"
+  url: ""
```
Comment (zaro0508) [inline config/prod/PsychENCODE-Globus-S3.yaml:4]: you don't need `templates/PEC-Globus-S3.j2`.  You just need to reference the [synapse-external-bucket.j2 file](https://github.com/Sage-Bionetworks/aws-infra/blob/master/templates/S3/synapse-external-bucket.j2)

Comment (zaro0508) [inline config/prod/PsychENCODE-Globus-S3.yaml:4]: url: "https://raw.githubusercontent.com/Sage-Bionetworks/aws-infra/v0.4.4/templates/S3/synapse-external-bucket.j2"

File: config/prod/PsychENCODE-Globus-S3.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  Project: "PsychENCODE"
+  OwnerEmail: 'dan.lu@sagebase.org'
+  CostCenter: "NO PROGRAM / 000000"
+parameters:
+  # The Sage deparment (Platform, CompOnc, SysBio, Governance, etc..)
+  Department: "SysBio"
+  # The Sage project (Infrastructure, amp-ad, ntap, dream, etc..)
+  Project: "PsychENCODE"
+  # The resource owner
+  OwnerEmail: 'dan.lu@sagebase.org'
```
Comment (zaro0508) [inline config/prod/PsychENCODE-Globus-S3.yaml:17]: these are no longer needed, please remove.

File: config/prod/PsychENCODE-Globus-S3.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  # The following parameters are only examples they are not required.
+  # You may omit them if you do not need to override the defaults.
+  # (Optional) true (default) to encrypt bucket, false for no encryption
+  EncryptBucket: 'false' # allow external users access to the bucket
```
Comment (zaro0508) [inline config/prod/PsychENCODE-Globus-S3.yaml:22]: We require buckets in scicomp to be encrypted.  It's set to encrypted by default so you can either remove this or comment it out.

File: templates/PEC-Globus-S3.j2 [hunk 4]
```diff
@@ -0,0 +1,216 @@
+AWSTemplateFormatVersion: 2010-09-09
```
[the remark this answers is not in the document]
Comment (danlu1) [reply templates/PEC-Globus-S3.j2:1]: I added s3:DeleteObject to the bucket policy. I saw there is a PR in aws-infra repo that might be ready to be merged. Can you check that?

File: config/prod/PsychENCODE-Globus-S3.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# Provision a Synapse External Bucket (http://docs.synapse.org/articles/custom_storage_location.html)
+template:
+  type: "http"
+  url: ""
+stack_name: "PsychENCODE-Globus-S3"
+stack_tags:
+  Department: "SysBio"
+  Project: "PsychENCODE"
+  OwnerEmail: 'dan.lu@sagebase.org'
+  CostCenter: "NO PROGRAM / 000000"
```
[the remark this answers is not in the document]
Comment (danlu1) [reply config/prod/PsychENCODE-Globus-S3.yaml:10]: I don't find an appropriate code for my data. Can I apply one?

[the remark this answers is not in the document]
Comment (zaro0508) [reply config/prod/PsychENCODE-Globus-S3.yaml:10]: those codes come from finance @danlu1.  I would talk to finance to get new codes.

[the remark this answers is not in the document]
Comment (danlu1) [reply config/prod/PsychENCODE-Globus-S3.yaml:10]: @zaro0508 Thanks!!

File: config/prod/PsychENCODE-Globus-S3.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Provision a Synapse External Bucket (http://docs.synapse.org/articles/custom_storage_location.html)
+template:
+  type: "http"
+  url: "https://raw.githubusercontent.com/Sage-Bionetworks/aws-infra/v0.4.4/templates/S3/synapse-external-bucket.j2"
```
Comment (zaro0508) [inline config/prod/PsychENCODE-Globus-S3.yaml:4]:
```suggestion
  url: "https://raw.githubusercontent.com/Sage-Bionetworks/aws-infra/v0.4.5/templates/S3/synapse-external-bucket.j2"
```

File: config/prod/PsychENCODE-Globus-S3.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  OwnerEmail: "dan.lu@sagebase.org"
+  CostCenter: "NO PROGRAM / 000000"
+parameters:
+  # The following parameters are only examples they are not required.
+  # You may omit them if you do not need to override the defaults.
+  # (Optional) true (default) to encrypt bucket, false for no encryption
+  # (Optional) true for read-write bucket, false (default) for read-only bucket
+  AllowWriteBucket: 'true'
+  # (Optional) Synapse username (default: ""), required if AllowWriteBucket=true
+  SynapseUserName: 'danlu'
```
Comment (zaro0508) [inline config/prod/PsychENCODE-Globus-S3.yaml:18]: we've replaced `SynapseUserName` with `SynapseIDs`.  please remove this line.

Please add this at the same level as `parameters:`
```
sceptre_user_data:
  SynapseIDs:
    - "3434599"
```

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: config/prod/PsychENCODE-Globus-S3.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (danlu1) [reply config/prod/PsychENCODE-Globus-S3.yaml:10]: @zaro0508 Thanks for you feedback, Khai. May I ask do we have an estimation time to get the cost center code? My director Mette is out this office, but she approved that we configure a separate S3 for this data transfer. If needed, I can share her approval message with finance team.

[the remark this answers is not in the document]
Comment (zaro0508) [reply config/prod/PsychENCODE-Globus-S3.yaml:10]: > May I ask do we have an estimation time to get the cost center code?

I'm not sure i understand your question @danlu1.  Finance is the one that can provide you with a cost center code.  IT has no control over that.  If it's taking too long we might be able to create a bucket now and apply a code retroactively but that might be a @brucehoff decision.

[the remark this answers is not in the document]
Comment (brucehoff) [reply config/prod/PsychENCODE-Globus-S3.yaml:10]: Khai, I believe Dan is saying that her cost center is not listed  here: [template/tags folder](https://github.com/Sage-Bionetworks/aws-infra/tree/master/templates/tags) Assuming that the code has to be in our GitHub doc's I have a request in to Finance to confirm the updated list, and will update our doc's.  If it doesn't have to be in the GitHub doc's for Dan to enter it in this PR, then please do so, @danlu1 .

[the remark this answers is not in the document]
Comment (danlu1) [reply config/prod/PsychENCODE-Globus-S3.yaml:10]: @brucehoff @zaro0508 Thanks guys, the cost center code is neither in [template/tags folder](https://github.com/Sage-Bionetworks/aws-infra/tree/master/templates/tags) nor  [2022July28ChartOfAccounts.csv](https://github.com/Sage-Bionetworks/scicomp-provisioner/files/9222334/2022July28ChartOfAccounts.csv). I will reach out to finance and discuss this.

**Review by thomasyu888 — APPROVED** [review]

File: config/prod/PsychENCODE-Globus-S3.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+parameters:
+  # The following parameters are only examples they are not required.
+  # You may omit them if you do not need to override the defaults.
+  # (Optional) true (default) to encrypt bucket, false for no encryption
+  # (Optional) true for read-write bucket, false (default) for read-only bucket
+  AllowWriteBucket: 'true'
+  # (Optional) Synapse username (default: ""), required if AllowWriteBucket=true
+  sceptre_user_data:
+  SynapseIDs:
+    - "3434599"
```
Comment (zaro0508) [inline config/prod/PsychENCODE-Globus-S3.yaml:20]: `sceptre_user_data:` needs to be at the same level as `parameters:`.  for example:

```
parameters:
  FirstParam: "foo"
  SecondParam: "bar"
sceptre_user_data:
  SynapseIDs:
    - "3434599"
```

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: config/prod/PsychENCODE-Globus-S3.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  # (Optional) Allow accounts, groups, and users to access bucket (default is no access).
+  GrantAccess:
+    - 'arn:aws:iam::325565585839:root'   # Required ARN for a synapse bucket
+    - 'arn:aws:sts::563295687221:assumed-role/AWSReservedSSO_Developer_baa6fed639faf5e7/dan.lu@sagebase.org'
```
Comment (zaro0508) [inline config/prod/PsychENCODE-Globus-S3.yaml:24]: please add `arn:aws:iam::531197219947:user/ucla-orion-transfer-user`

File: config/prod/PsychENCODE-Globus-S3.yaml [hunk 10]
```diff
@@ -15,10 +15,11 @@ parameters:
   # (Optional) true for read-write bucket, false (default) for read-only bucket
   AllowWriteBucket: 'true'
   # (Optional) Synapse username (default: ""), required if AllowWriteBucket=true
-  sceptre_user_data:
+sceptre_user_data:
   SynapseIDs:
     - "3434599"
```
Comment (zaro0508) [inline config/prod/PsychENCODE-Globus-S3.yaml:20]: GrantAccess goes with parameter, not sceptre_user_data.  Please move this section to the bottom of the file.

**Review by zaro0508 — CHANGES_REQUESTED** [review]

**Review by zaro0508 — APPROVED** [review]

**Pull request merged**